build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/androidhomewarmup/AndroidHomeWarmupTask.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.integrationtests.androidhomewarmup
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.provider.Provider
+import org.gradle.process.ExecOperations
+import org.gradle.process.ExecResult
+import java.io.File
+import javax.inject.Inject
+import org.gradle.work.DisableCachingByDefault
+
+@DisableCachingByDefault(because = "Not worth caching")
```

Comment:
💅 this task must not be cached, because the outputs are not what actually matters